tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,452 @@
+import pytest
+import json
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+from sklearn import datasets
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as spark_f
+from pyspark.ml.connect.classification import (
+    LogisticRegression as LORV2,
+)
+from pyspark.ml.connect.feature import StandardScaler
+from pyspark.ml.connect.pipeline import Pipeline
+from pyspark.sql.types import LongType
+
+import os
+from unittest import mock
+import yaml
+import mlflow
+from mlflow import pyfunc
+from mlflow import spark as sparkm
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+import mlflow.pyfunc.scoring_server as pyfunc_scoring_server
+from tests.pyfunc.test_spark import score_model_as_udf
+from tests.spark.test_spark_model_export import (
+    SparkModelWithData,
+)
+from mlflow.pyfunc import spark_udf
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from tests.helper_functions import (
+    _assert_pip_requirements,
+    _compare_conda_env_requirements,
+    _compare_logged_code_paths,
+    _mlflow_major_version_string,
+    assert_register_model_called_with_local_model_path,
+    pyfunc_serve_and_score_model,
+)
+from mlflow.utils.file_utils import TempDir
+
+
+import pyspark
+from packaging.version import Version
+
+if Version(pyspark.__version__) < Version("3.5"):
+    pytest.skip("pyspark ML connect Model is available in pyspark >= 3.5")
+
+
+def _get_spark_connect_session():
+    return (
+        SparkSession.builder.remote("local[2]")
+            .config("spark.connect.copyFromLocalToFs.allowDestLocal", "true")
+            .getOrCreate()
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+@pytest.fixture
+def spark_connect_model_custom_env(tmp_path):
+    conda_env = os.path.join(tmp_path, "conda_env.yml")
+    _mlflow_conda_env(conda_env, additional_pip_deps=[
+        "https://dist.apache.org/repos/dist/dev/spark/v3.5.0-rc4-bin/pyspark-3.5.0.tar.gz",
```

Comment:
Can we add a ticket to update this after 3.5 release?

Reply:
Sure :)

Reply:
Ticket: https://databricks.atlassian.net/browse/ML-34786